src/layout/dom.ts

```diff
@@ -41,15 +59,20 @@ export function add(node: Node, data: NodeInfo, source: Source): void {
     let nextId = getNextId(node);
     let masked = true;
     let parent = null;
+    let region = regionMap.has(node) ? regionMap.get(node) : null;
 
     if (parentId >= 0 && values[parentId]) {
         parent = values[parentId];
         parent.children.push(id);
+        region = region === null ? parent.region : region;
```

Comment:
>region [](start = 42, length = 6)

is this guarantee to exist or do you need to check?

Reply:
Yes, the value can be null but it is guaranteed to exist.